### PR TITLE
[SYCLomatic] Remove redundant -c flag

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Ahmed, Daiyaan <daiyaan.ahmed@intel.com>

### CODE REVIEW

File: clang/tools/scan-build-py/lib/libscanbuild/intercept.py [hunk 1]
```diff
@@ -294,9 +294,12 @@ def abspath(cwd, name):
             file_path = abspath(exec_trace['directory'], preprocess_source)
             logging.debug('preprocess_file: %s,', file_path)
             preproess_output_files.add(file_path)
-
+
```
Comment (ShengchenJ) [inline clang/tools/scan-build-py/lib/libscanbuild/intercept.py]: Remove the unnecessarily space

Comment (daiyaan-ahmed6) [reply clang/tools/scan-build-py/lib/libscanbuild/intercept.py]: Removed.

**Review by ShengchenJ — APPROVED** [review]
LGTM

**Review by tomflinda — APPROVED** [review]
LGTM, pls make sure the behavior test can verify this PR.

File: clang/tools/scan-build-py/lib/libscanbuild/intercept.py [hunk 2]
```diff
@@ -294,9 +294,11 @@ def abspath(cwd, name):
             file_path = abspath(exec_trace['directory'], preprocess_source)
             logging.debug('preprocess_file: %s,', file_path)
             preproess_output_files.add(file_path)
-
         for source in compilation.files:
-            command = [compiler, "-c"] + compilation.flags + [source]
+            command = [compiler]
+            if('-o' not in compilation.flags):
```
Comment (ziranzha) [inline clang/tools/scan-build-py/lib/libscanbuild/intercept.py:299]: Option `-o` is not related to option `-c`.
Please ask @tomflinda for more details.

Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libscanbuild/intercept.py:299]: @daiyaan-ahmed6  as discussed  with you and @ShengchenJ, this PR is not needed, pls close it.

**Review by tomflinda — CHANGES_REQUESTED** [review]
-1

**Pull request closed without merging**